pandas/tests/test_generic.py

```diff
@@ -872,6 +872,14 @@ def test_interp_limit_no_nans(self):
         expected = s
         assert_series_equal(result, expected)
 
+    def test_interp_piecewise_polynomial(self):
+        # GH 10365
+        tm._skip_if_no_scipy()
+        s = Series([1, 2, 3, 4, np.nan, 6, np.nan])
+        result = s.interpolate(method='piecewise_polynomial')
+        expected = Series([1., 2., 3., 4., 5., 6., 7.])
+        assert_series_equal(expected, result)
```

Comment:
can you add some tests with nans up front? (maybe for several other interpolate methods as well to make sure nothing breaking)